src/guidellm/benchmark/benchmark.py

```diff
@@ -0,0 +1,796 @@
+    @property
+    def output_tokens_per_second(self) -> Optional[float]:
+        """
+        :return: The average number of tokens generated per second in the output text.
+            Note, does not include the time to generate the first token. None if there
+            were no output_tokens or the first token was not received.
+        """
+        if self.inter_token_latency_ms is None:
+            return None
+
+        return super().output_tokens_per_second
+
+
+class GenerativeBenchmark(Benchmark):
+    """
+    A serializable model representing a benchmark run and its results for generative
+    requests and responses. Includes the completed and errored requests, the start
+    and end times for the benchmark, and the statistics for the requests and responses.
+    """
+
+    successful_total: int = Field(
+        description=(
+            "The total number of completed requests in the benchmark, "
+            "excluding warmup and cooldown."
+        )
+    )
+    successful_sampled_size: Optional[int] = Field(
+        default=None,
+        description=(
+            "The number of completed requests that were randomly sampled for "
+            "the benchmark. None if no sampling was applied."
+        ),
+    )
+    successful_requests: List[GenerativeTextResponseStats] = Field(
+        description="The list of completed requests.",
+    )
+    incomplete_total: int = Field(
+        description=(
+            "The total number of incomplete requests in the benchmark, "
+            "excluding warmup and cooldown."
+        )
+    )
+    incomplete_sampled_size: Optional[int] = Field(
+        default=None,
+        description=(
+            "The number of incomplete requests that were randomly sampled for "
+            "the benchmark. None if no sampling was applied."
+        ),
+    )
+    incomplete_requests: List[GenerativeTextResponseStats] = Field(
+        description="The list of incomplete requests.",
+    )
+    errored_total: int = Field(
+        description=(
+            "The total number of errored requests in the benchmark, "
+            "excluding warmup and cooldown."
+        )
+    )
+    errored_sampled_size: Optional[int] = Field(
+        default=None,
+        description=(
+            "The number of errored requests that were randomly sampled for "
+            "the benchmark. None if no sampling was applied."
+        ),
+    )
+    errored_requests: List[GenerativeTextErrorStats] = Field(
+        description="The list of errored requests.",
+    )
+    start_time: float = Field(
+        description="The start time of the first request for the benchmark.",
+    )
+    end_time: float = Field(
+        description="The end time of the last request for the benchmark.",
+    )
+
+    requests_latency: StatusDistributionSummary = Field(
+        description="The distribution of latencies for the completed requests.",
+    )
+    prompts_token_count: StatusDistributionSummary = Field(
```

Comment:
nit, but I think this sounds less awkward for the name of the distribution
```suggestion
    prompt_token_count: StatusDistributionSummary = Field(
```